File: app/src/main/java/com/androidskeleton/module/main/MainModule.java
package com.androidskeleton.module.main;

import com.androidskeleton.di.scope.FragmentScoped;
import com.androidskeleton.di.scope.PerActivity;
import com.androidskeleton.module.ui.home.HomeContract;
import com.androidskeleton.module.ui.home.HomeFragment;
import com.androidskeleton.module.ui.home.HomePresenter;

import dagger.Binds;
import dagger.Module;
import dagger.android.ContributesAndroidInjector;

@Module
public abstract class MainModule {
    @FragmentScoped
    @ContributesAndroidInjector
    abstract HomeFragment mainFragment();

    @PerActivity
    @Binds
    abstract HomeContract.Presenter mainPresenter(HomePresenter presenter);
}

File: app/src/main/java/com/androidskeleton/module/ui/main/MainModule.kt
package com.androidskeleton.module.ui.main

import com.androidskeleton.di.scope.FragmentScoped
import com.androidskeleton.di.scope.PerActivity
import com.androidskeleton.module.ui.home.HomeContract.Presenter
import com.androidskeleton.module.ui.home.HomeFragment
import com.androidskeleton.module.ui.home.HomePresenter
import dagger.Binds
import dagger.Module
import dagger.android.ContributesAndroidInjector

@Module
abstract class MainModule {
    @FragmentScoped
    @ContributesAndroidInjector
    abstract fun mainFragment(): HomeFragment?

    @PerActivity
    @Binds
    abstract fun mainPresenter(presenter: HomePresenter?): Presenter?
}